test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -0,0 +1,235 @@
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test").
+		Body(yamlBody).
+		DoRaw()
+	if err != nil {
+		t.Fatal(err, string(result))
+	}
+
+	result, err = rest.Patch(types.MergePatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Body([]byte(`{"spec":{"replicas": 5}}`)).
+		DoRaw()
+	if err != nil {
+		t.Fatal(err, string(result))
+	}
+
+	// Re-apply the same object, we should get conflicts now.
+	result, err = rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test").
+		Body(yamlBody).
+		DoRaw()
+	if err == nil {
+		t.Fatalf("Expecting to get conflicts when applying object, got no error: %s", result)
+	}
+	status, ok := err.(*errors.StatusError)
+	if !ok {
+		t.Fatalf("Expecting to get conflicts as API error")
+	}
+	if len(status.Status().Details.Causes) < 1 {
+		t.Fatalf("Expecting to get at least one conflict when applying object, got: %v", status.Status().Details.Causes)
+	}
+
+	// Re-apply with force, should work fine.
+	result, err = rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("force", "true").
+		Param("fieldManager", "apply_test").
+		Body(yamlBody).
+		DoRaw()
+	if err != nil {
+		t.Fatal(err, string(result))
+	}
+
+	// New applier edit existing list item, we should get conflicts.
+	result, err = rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test_2").
+		Body([]byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+spec:
+  ports:
+  - name: "y"
+    containerPort: 80
+    protocol: TCP`, apiVersion, kind, name))).
+		DoRaw()
+	if err == nil {
+		t.Fatalf("Expecting to get conflicts when applying object, got no error: %s", result)
+	}
+	status, ok = err.(*errors.StatusError)
+	if !ok {
+		t.Fatalf("Expecting to get conflicts as API error")
+	}
+	if len(status.Status().Details.Causes) < 1 {
+		t.Fatalf("Expecting to get at least one conflict when applying object, got: %v", status.Status().Details.Causes)
+	}
+
+	// New applier add list item, should work fine.
+	result, err = rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test_2").
+		Body([]byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+spec:
+  ports:
+  - name: "y"
+    containerPort: 8080
+    protocol: TCP`, apiVersion, kind, name))).
+		DoRaw()
+	if err != nil {
+		t.Fatal(err, string(result))
+	}
```

Comment:
we need a test that checks that apply patches do not work for CRs without structural schema.